jwst/datamodels/container.py

```diff
@@ -477,15 +478,11 @@ def ind_asn_type(self, asn_exptype):
         ind : list
             Indices of models in ModelContainer._models matching ``asn_exptype``.
         """
-        names = []
-        for m in self.meta.asn_table.products[0].members:
-            if m.exptype.lower() == asn_exptype:
-                file_path, file_name = op.split(m.expname)
-                names.append(file_name)
         ind = []
         for i, model in enumerate(self._models):
-            if model.meta.filename in names:
-                ind.append(i)
+            if model.meta.asn.exptype:
+                if model.meta.asn.exptype.lower() == asn_exptype:
```

Comment:
I feel like there must be a better way of doing this - checking to ensure the entry exists, then applying .lower() to the result if so. Any thoughts? I was envisioning something like a dict get() method that would provide an empty string if no value was present, but maybe I'm overthinking it.

Reply:
if there is a chance this does not exist, then the `get` is the way to go. As for the default, maybe the default should be `science`, since, at this point, if there is no knowledge, the presumption is that any data is science.

Reply:
Sorry to have left this sitting, going through the Help Desk wilderness...

Where exactly can get be applied here? I tried something like model.get('meta.asn.exptype'), but get an AttributeError from the datamodel having no 'get' method. Setting the default to science does seem appropriate.